File: Remove_Outermost_Parentheses.py
s = "(()())(())(()(()))"

def removeOuterParentheses(self, s)
    res = ""
    c = 0
    for i in range(len(s)):
        if s[i] == "(":
            c+=1
            if c > 1:
                res +="("
        else:
            if c > 1:
                res +=")"
            c -= 1
                
    return res


print(removeOuterParentheses(s))

File: Remove_Outermost_Parentheses/Remove_Outermost_Parentheses.go
package main

import "fmt"

func main() {
	s := "(()())(())(()(()))"
	fmt.Println(removeOuterParentheses(s))

}

func removeOuterParentheses(s string) string {
	var res string
	c := 0
	for _, v := range s {
		if string(v) == "(" {
			c++
			if c > 1 {
				res += "("
			}
		} else {
			if c > 1 {
				res += ")"
			}
			c -= 1
		}
	}
	return res
}
